src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/ShippingMethod.xml

```diff
@@ -0,0 +1,286 @@
+            </operation>
+
+            <operation
+                class="ApiPlatform\Metadata\Patch"
+                uriTemplate="/admin/shipping-methods/{code}/archive"
+                input="false"
+                controller="Sylius\Bundle\ApiBundle\Applicator\ArchivingShippingMethodApplicatorInterface::archive"
+            >
+                <normalizationContext>
+                    <values>
+                        <value name="groups">
+                            <values>
+                                <value>sylius:admin:shipping_method:show</value>
+                            </values>
+                        </value>
+                    </values>
+                </normalizationContext>
+                <openapiContext>
+                    <values>
+                        <value name="summary">Archives Shipping Method.</value>
+                    </values>
+                </openapiContext>
+            </operation>
+
+            <operation
+                class="ApiPlatform\Metadata\Patch"
+                uriTemplate="/admin/shipping-methods/{code}/restore"
+                input="false"
+                controller="Sylius\Bundle\ApiBundle\Applicator\ArchivingShippingMethodApplicatorInterface::restore"
+            >
+                <normalizationContext>
+                    <values>
+                        <value name="groups">
+                            <values>
+                                <value>sylius:admin:shipping_method:show</value>
+                            </values>
+                        </value>
+                    </values>
+                </normalizationContext>
+                <openapiContext>
+                    <values>
+                        <value name="summary">Restores archived Shipping Method.</value>
+                    </values>
+                </openapiContext>
+            </operation>
+
+            <operation
+                class="ApiPlatform\Metadata\Delete"
+                uriTemplate="/admin/shipping-methods/{code}"
+                processor="sylius_api.state_processor.admin.shipping_method.remove"
+            />
+
+            <!-- Shop operations-->
+
+            <operation
+                class="ApiPlatform\Metadata\GetCollection"
+                uriTemplate="/shop/shipping-methods"
+                provider="sylius_api.state_provider.shop.shipping_method.collection"
+                paginationEnabled="false"
+            >
+                <normalizationContext>
+                    <values>
+                        <value name="groups">
+                            <values>
+                                <value>sylius:shop:shipping_method:index</value>
+                            </values>
+                        </value>
+                    </values>
+                </normalizationContext>
+            </operation>
+
+            <operation class="ApiPlatform\Metadata\Get"  uriTemplate="/shop/shipping-methods/{code}">
```

Comment:
In the collectionOperation, we have a provider, so we get the shipping method only for the channel. However, in this itemOperation, we don't have this restriction, so with the code, I can get information about each shipping method.